.github/workflows/desktop.yml

```diff
@@ -0,0 +1,160 @@
+          brew install ccache
+
+      - name: Install system level applications (linux)
+        if: startsWith(matrix.os, 'ubuntu')
+        id: ubuntu-brew-install
+        run: |
+          sudo apt install ccache
+
+      - name: Cache vcpkg C++ dependencies
+        id: cache_vcpkg
+        uses: actions/cache@v2
+        with:
+          path: external/vcpkg/installed
+          key: dev-vcpkg-${{ matrix.vcpkg_triplet }}-${{ hashFiles(format('{0}', env.VCPKG_RESPONSE_FILE)) }}-${{ hashFiles('.git/modules/external/vcpkg/HEAD') }}
+
+      - name: Install vcpkg packages (windows)
+        if: steps.cache_vcpkg.outputs.cache-hit!='true' && startsWith(matrix.os, 'windows')
+        run: |
+          ${{ github.workspace }}/external/vcpkg/bootstrap-vcpkg.bat -disableMetrics
+          ${{ github.workspace }}/external/vcpkg/vcpkg install @${{env.VCPKG_RESPONSE_FILE}}
+
+      - name: Install vcpkg packages (mac and linux)
+        if: steps.cache_vcpkg.outputs.cache-hit!='true' && startsWith(matrix.os, 'windows')!='true'
+        run: |
+          ${{ github.workspace }}/external/vcpkg/bootstrap-vcpkg.sh -disableMetrics
+          ${{ github.workspace }}/external/vcpkg/vcpkg install @${{env.VCPKG_RESPONSE_FILE}}
+
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python_version }}
+          architecture: ${{ matrix.architecture }}
+
+      - name: Get pip cache dir
+        id: get_pip_cache_dir
+        run: |
+          echo "::set-output name=cachedir::$(pip cache dir)"
+
+      - name: pip cache
+        id: cache_pip
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.get_pip_cache_dir.outputs.cachedir }}
+          key: dev-pip-${{ runner.os }}-${{ matrix.python_version }}-${{ matrix.architecture }}-${{ hashFiles(format('{0}', env.PIP_REQUIREMENTS_FILE)) }}
+
+      - name: Install pip packages
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r ${{ env.PIP_REQUIREMENTS_FILE }}
+
+      - name: Cache ccache files (mac and linux)
+        if: startsWith(matrix.os, 'macos') || startsWith(matrix.os, 'ubuntu')
+        id: cache_ccache
+        uses: actions/cache@v2
+        with:
+          path: ccache_dir
+          key: dev-test-ccache-${{ env.MATRIX_UNIQUE_NAME }}
+
+      - name: Configure 
+        shell: bash
+        run: |
+          mkdir build
+          cmake \
+            -S . \
+            -B build \
+            -G "${{ matrix.generator }}" \
+            -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
+            -DCMAKE_TOOLCHAIN_FILE=external/vcpkg/scripts/buildsystems/vcpkg.cmake \
+            -DVCPKG_TARGET_TRIPLET=${{ matrix.vcpkg_triplet }} \
+
+      - name: Build
+        shell: bash
+        run: cmake --build build --config ${{ matrix.build_type }} -j 8
+
+      - name: Stats for ccache (mac and linux)
+        if: startsWith(matrix.os, 'macos') || startsWith(matrix.os, 'ubuntu')
```

Comment:
Instead of checking for the OSes here and assuming that cmake is enabled / installed, maybe make the "- name: Install system level applications (linux)" and "- name: Install system level applications (macos)" steps (above) set a flag indicating that cmake WAS installed and that this check to just be for that flag.  Is that possible?

Reply:
Excellent point. Yes, let me look into this. I am certain I can set a global environment variable during the installation but I am not sure if I can use that in an "if" expression. Going to give this a try.

Reply:
So this experiment was successful.

```
        # Setting an env variable right after installing ccache that can be reused in later stages

      - name: Install system level applications (linux)
        if: startsWith(matrix.os, 'ubuntu')
        id: ubuntu-apt-install
        run: |
          sudo apt install ccache
          echo "::set-env name=CCACHE_INSTALLED::1"

      - name: Cache ccache files
        if: env.CCACHE_INSTALLED
        id: cache_ccache
```